include/wlr/types/wlr_buffer.h

```diff
@@ -69,6 +69,11 @@ struct wlr_buffer {
 	} events;
 };
 
+struct wlr_buffer_resource_impl {
+	bool (*is_instance)(struct wl_resource *resource);
```

Comment:
Maybe we could add a `const char *name` field here as well? That would make debugging easier, e.g. we could print the kind of the buffer resource in the error messages in the next commit, instead of just referring to a "custom buffer".

Reply:
Sure!